src/aioquic/quic/congestion/congestion.py

```diff
@@ -0,0 +1,155 @@
+from ..packet_builder import QuicSentPacket
+from typing import Iterable, Optional, Dict, Any
+from datetime import datetime
+from enum import Enum
+
+K_GRANULARITY = 0.001  # seconds
+
+# congestion control
+K_INITIAL_WINDOW = 10
+K_MINIMUM_WINDOW = 2
+K_LOSS_REDUCTION_FACTOR = 0.5
+
+class CongestionEvent(Enum):
+    ACK=0
+    PACKET_SENT=1
+    PACKET_EXPIRED=2
+    PACKET_LOST=3
+    RTT_MEASURED=4
+
+class QuicRttMonitor:
+    """
+    Roundtrip time monitor for HyStart.
+    """
+
+    def __init__(self) -> None:
+        self._increases = 0
+        self._last_time = None
+        self._ready = False
+        self._size = 5
+
+        self._filtered_min: Optional[float] = None
+
+        self._sample_idx = 0
+        self._sample_max: Optional[float] = None
+        self._sample_min: Optional[float] = None
+        self._sample_time = 0.0
+        self._samples = [0.0 for i in range(self._size)]
+
+    def add_rtt(self, rtt: float) -> None:
+        self._samples[self._sample_idx] = rtt
+        self._sample_idx += 1
+
+        if self._sample_idx >= self._size:
+            self._sample_idx = 0
+            self._ready = True
+
+        if self._ready:
+            self._sample_max = self._samples[0]
+            self._sample_min = self._samples[0]
+            for sample in self._samples[1:]:
+                if sample < self._sample_min:
+                    self._sample_min = sample
+                elif sample > self._sample_max:
+                    self._sample_max = sample
+
+    def is_rtt_increasing(self, rtt: float, now: float) -> bool:
+        if now > self._sample_time + K_GRANULARITY:
+            self.add_rtt(rtt)
+            self._sample_time = now
+
+            if self._ready:
+                if self._filtered_min is None or self._filtered_min > self._sample_max:
+                    self._filtered_min = self._sample_max
+
+                delta = self._sample_min - self._filtered_min
+                if delta * 4 >= self._filtered_min:
+                    self._increases += 1
+                    if self._increases >= self._size:
+                        return True
+                elif delta > 0:
+                    self._increases = 0
+        return False
+    
+
+def Now():
+    return datetime.timestamp(datetime.now())
```

Comment:
I'm not sure I'm comfortable with this, until now all timestamps have been provided as input to the congestion controller's method. 

If I remember correctly, this timestamp is handed to us by the event loop, and there is no guarantee its aligns with `datetime.now`.